### PR TITLE
Adding ability to have the executor not create a namespace.

### PR DESCRIPTION
**Pull request by shawn-hurley**
* Adding Executor configuration struct.
* Changing the signature of new executor

### CODE REVIEW

**Review by mhrivnak — APPROVED** [review]
I hadn't noticed how we have nearly the same 80 lines of code repeated in 4 places. That's way beyond the scope of this PR, but would be interesting to chat about another time.

**Review by dymurray — APPROVED** [review]

File: bundle/bind.go [hunk 1]
```diff
@@ -44,8 +44,12 @@ func (e *executor) Bind(
 
 	go func() {
 		e.actionStarted()
-		// Create namespace name that will be used to generate a name.
-		ns := fmt.Sprintf("%s-%.4s-", instance.Spec.FQName, bindAction)
+		ns := instance.Context.Namespace
+		// Determine if we should be using the context namespace from the executor config.
+		if !e.doNotCreateNS {
```
Comment (eriknelson) [inline bundle/bind.go]: This is confusing phrasing with the double negative. I would do:

```
if e.doNotCreateNS {
ns := instance.Context.Namespace
else {
ns := createdNamespace
}
```

Less mental gymnastics and more direct.

**Review by eriknelson — CHANGES_REQUESTED** [review]
Comment applies to all the conditional ns set statements, otherwise looks good.

File: bundle/bind.go (hunk 1 above)
Comment (jmrodri) [reply bundle/bind.go]: I prefer either: `createNS` OR `skipCreateNS`. Using skip allows you to use the Golang default of False for bools and to me it's easier to read then `doNot`

**Review by jmrodri — CHANGES_REQUESTED** [review]
fix the double negative, otherwise +1

**Review by jmrodri — APPROVED** [review]

**Pull request merged**